beater/beater.go

```diff
@@ -1026,3 +1027,23 @@ func queryClusterUUID(ctx context.Context, esClient elasticsearch.Client) error
 	s.Set(response.ClusterUUID)
 	return nil
 }
+
+func adjustMaxProcs(ctx context.Context, d time.Duration, logger *logp.Logger) error {
+	setMaxProcs := func() {
+		if _, err := maxprocs.Set(maxprocs.Logger(logger.Infof)); err != nil {
```

Comment:
`maxprocs.Set` logs something every time it is called, which may be quite noisy. Do you think it would be reasonable to log to a buffer, and only send that to `logger.Info` on the first call or otherwise if runtime.GOMAXPROCS changes?

Reply:
Great point, I think that's a much better solution. I'll make the changes.